src/Arduboy.cpp

```diff
@@ -385,10 +385,47 @@ void Arduboy::drawFastVLine
 void Arduboy::drawFastHLine
 (int16_t x, int16_t y, uint8_t w, uint8_t color)
 {
-  int end = x+w;
-  for (int a = max(0,x); a < min(end,WIDTH); a++)
+  // Do bounds/limit checks
+  if (y < 0 || y >= HEIGHT) {
+    return;
+  }
+
+  // make sure we don't try to draw below 0
+  if (x < 0) {
+    w += x;
+    x = 0;
+  }
+
+  // make sure we don't go off the edge of the display
+  if ((x + w) > WIDTH) {
+    w = (WIDTH - x);
+  }
+
+  // if our width is now negative, punt
+  if (w <= 0) {
+    return;
+  }
+
+  // buffer pointer plus row offset + x offset
+  register uint8_t *pBuf = sBuffer + ((y/8) * WIDTH) + x;;
```

Comment:
If you mean the double semicolon, it's probably a typo. The second semicolon will just be a null statement and do nothing. It will compile correctly and produce exactly the same code as if there was only one semicolon.

It should probably be fixed, though.